todoapp/src/components/TodoUl.vue

```diff
@@ -0,0 +1,33 @@
+<script setup lang="ts">
+import { defineProps } from 'vue'
+const props = defineProps(['todos'])
+const emits = defineEmits(['removeTodo'])
+
+function removeTodo(todo: string) {
+  emits('removeTodo', todo)
+}
+</script>
+
+<template>
+  <ul class="todo-ul">
+    <li class="todo" v-for="todo in todos" :key="todo.text">
```

Comment:
Please strive for the key to be unique, in this case the key on the v-for could be similar, I would like to ask you to also concatenate the index of the todo with the text, so we have a higher probability of the key being unique. 
This might not matter much in your current implementation, but it will matter when you work with more complex data. Since having an unique key is crucial for the rendering engine to be able to know the difference between each node in the dom, so it can update them properly when they change.